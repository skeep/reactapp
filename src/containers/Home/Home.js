import React, { Component } from 'react'
import autoBind from 'react-autobind'
import { connect } from 'react-redux'
import { Helmet } from 'react-helmet'

import Actions from 'Actions'
import Selectors from 'Selectors'

import { LiveSpan } from 'Components'

class Home extends Component {
  constructor (props) {
    super(props)
    autoBind(this)
  }

  componentDidMount () {
    this.props.dispatch(Actions.Beers.fetchBeer())
  }

  render () {
    const _beers = this.props.beers
    const _name = _beers.payload && _beers.payload.length ? _beers.payload[0].name : null
    return (
      <div>
        <Helmet>
          <title>Home</title>
        </Helmet>
        <h2>
          Home :
          <LiveSpan
            payload={_name}
            error={_beers.error}
            fetching={_beers.meta.fetching}
            message={_beers.meta.message}
          />
        </h2>
      </div>
    )
  }
}

function mapStateToProps (state) {
  return {
    beers: Selectors.Beers.getBeers(state)
  }
}

export default connect(mapStateToProps)(Home)
