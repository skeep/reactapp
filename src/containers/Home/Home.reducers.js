const HomeState = (state = {}, action) => {
  switch (action.type) {
    default:
      return state
  }
}

export default HomeState
