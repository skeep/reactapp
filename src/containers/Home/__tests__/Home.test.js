import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import Home from '../Home.js'

// Snapshot test
it('Snapshot test with content', () => {
  const tree = shallow(<Home />)
  expect(toJson(tree)).toMatchSnapshot()
})

// Rendering test
it('Rendering test with content', () => {
  const tree = shallow(<Home />)
  expect(tree.find('h2').text()).toEqual('Home')
})

// Behaviour tests

// Integration tests

// System tests
