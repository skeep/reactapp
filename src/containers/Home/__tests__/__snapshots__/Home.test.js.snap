// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test with content 1`] = `
<div>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      Home
    </title>
  </HelmetWrapper>
  <h2>
    Home
  </h2>
</div>
`;
